apps/design/frontend/src/encode_metadata.ts

```diff
@@ -0,0 +1,115 @@
+import { assert, iter } from '@votingworks/basics';
+
+type Bit = 0 | 1;
+
+function numberToBitsRtl(num: number, bitLength: number): Bit[] {
+  return num
+    .toString(2)
+    .padStart(bitLength, '0')
+    .split('')
+    .map((char) => (char === '0' ? 0 : 1))
+    .reverse();
+}
+
+/**
+ * Encodes ballot metadata as the bottom row of timing marks on a ballot.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function encodeMetadata(ballotStyleIndex: number) {
+  const ballotStyleId = `card-number-${ballotStyleIndex}`;
+
+  const frontMetadataWithoutChecksum = {
+    /**
+     * Mod 4 check sum from bits 0-1 (2 bits).
+     *
+     * The mod 4 check sum bits are obtained by adding the number of 1’s in bits 2
+     * through 31, then encoding the results of a mod 4 operation in bits 0 and 1.
+     * For example, if bits 2 through 31 have 18 1’s, bits 0 and 1 will hold the
+     * value 2 (18 mod 4 = 2).
+     */
+    // mod4Checksum: u8;
+
+    /** Batch or precinct number from bits 2-14 (13 bits). */
+    batchOrPrecinctNumber: {
+      value: 0, // Unused, arbitrary
+      bitLength: 13,
+    },
+
+    /** Card number (CardRotID) from bits 15-27 (13 bits). */
+    cardNumber: {
+      value: ballotStyleIndex,
+      bitLength: 13,
+    },
+
+    /** Sequence number (always 0) from bits 28-30 (3 bits). */
+    sequenceNumber: {
+      value: 0, // Unused
+      bitLength: 3,
+    },
+
+    /** Start bit (always 1) from bit 31-31 (1 bit). */
+    startBit: {
+      value: 1,
+      bitLength: 1,
+    },
+  } as const;
```

Comment:
Is this format (start bit + sequenceNumber + cardNumber + batchOrPrecintNumber) also a convention? (Same question for the back metadata)

Reply:
Same answer as above